### PR TITLE
x-extension-path-converter-bugfix

### PR DESCRIPTION
**Pull request by qiaozha**

### CODE REVIEW

File: src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs [hunk 1]
```diff
@@ -26,13 +26,13 @@ public override bool CanConvert(System.Type objectType)
             JsonSerializer serializer)
         {
             // is the leaf an vendor extension? "x-*..."
-            if (reader == null || reader.Path.Substring(reader.Path.LastIndexOf(".", StringComparison.Ordinal) + 1).StartsWith("x-", StringComparison.CurrentCulture))
+            JObject jobject = JObject.Load(reader);
```
Comment (ruowan) [inline src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs]: You can add some unit tests to cover the new feature.

Comment (ruowan) [reply src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs]: You can add some comments here to list which case is invalid.

File: src/dotnet/OpenAPI.Validator.Tests/Resource/OpenAPI/Validation/positive/x-ms-paths-with-odata.json [hunk 2]
```diff
@@ -0,0 +1,1354 @@
+{
```
Comment (ruowan) [inline src/dotnet/OpenAPI.Validator.Tests/Resource/OpenAPI/Validation/positive/x-ms-paths-with-odata.json]: Could you pls simplify the test fixtures? I think not all data is required.

File: src/dotnet/OpenAPI.Validator.Tests/Resource/OpenAPI/Validation/positive/x-ms-paths-with-odata.json [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+      "name": "Authorization",
+      "in": "header"
+    }
+  },
+  "schemes": ["https"],
+  "consumes": ["application/xml"],
+  "produces": ["application/xml"],
+  "paths": {},
+  "x-ms-paths": {
+    "/{table}(PartitionKey='{partitionKey}',RowKey='{rowKey}')": {
```
Comment (qiaozha) [inline src/dotnet/OpenAPI.Validator.Tests/Resource/OpenAPI/Validation/positive/x-ms-paths-with-odata.json]: Previously this line will cause swagger.parse fail

File: src/dotnet/OpenAPI.Validator.Tests/OpenAPIModelerValidationTests.cs [hunk 4]
```diff
@@ -890,6 +890,23 @@ public void ValidExtensionForLongRunningOperationOptionsForLROPostWithNoResponse
...
+        
+        [Fact]
+        public void ValidXMSPathWithOAata() 
+        {
+            var filePath = Path.Combine(AutoRest.Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "OpenAPI", "Validation", "positive", "x-ms-paths-with-odata.json");
+            var fileText = File.ReadAllText(filePath);
+            var message = "";
+            try
+            {
+                var servDef = SwaggerParser.Parse(filePath, fileText);
```
Comment (qiaozha) [inline src/dotnet/OpenAPI.Validator.Tests/OpenAPIModelerValidationTests.cs]: this tests if the SwaggerParser can Parse src/dotnet/OpenAPI.Validator.Tests/Resource/OpenAPI/Validation/positive/x-ms-paths-with-odata.json correctly.

**Comment by ruowan** [comment]
Pls update the changelog file. https://github.com/Azure/azure-openapi-validator/blob/master/changelog.md

**Comment by qiaozha** [comment]
> Pls update the changelog file. https://github.com/Azure/azure-openapi-validator/blob/master/changelog.md

add change log. pls check

File: src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs (hunk 1 above)
Comment (ruowan) [reply src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs]: Why you move `JObject jobject ...`this line from bottom to here.

Comment (qiaozha) [reply src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs]: That's the line causing the problem. it return too soon that the parse process doesn't move on, causing it runs the same step next time. and  causing next step has null pointer error.

**Pull request merged**